roles/kubernetes-apps/container_engine_accelerator/nvidia_gpu/templates/k8s-device-plugin-nvidia-daemonset.yml.j2

```diff
@@ -0,0 +1,61 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: nvidia-gpu-device-plugin
+  namespace: kube-system
+  labels:
+    k8s-app: nvidia-gpu-device-plugin
+    addonmanager.kubernetes.io/mode: Reconcile
```

Comment:
Does kubespray deploy the addon manager? If not then we should probably eliminate this

Reply:
I'll have to check to this.

Reply:
So i don't think the addon manager is deployed by kubespray. But checking the usage of `addonmanager.kubernetes.io/mode` in the kubespray project it is used pretty much in each `addon`. This also comes default with the `k8s-device-plugin-nvidia-daemon` from GCP. 
```SHELL
 grep -nr addonmanager * 
roles/dnsmasq/templates/dnsmasq-autoscaler.yml.j2:24:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/container_engine_accelerator/nvidia_gpu/templates/k8s-device-plugin-nvidia-daemonset.yml.j2:8:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/registry/templates/registry-pvc.yml.j2:9:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/registry/templates/registry-rs.yml.j2:11:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/registry/templates/registry-svc.yml.j2:10:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/ansible/templates/kubedns-autoscaler.yml.j2:24:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/ansible/templates/coredns-deployment.yml.j2:10:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/ansible/templates/coredns-clusterrole.yml.j2:7:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/ansible/templates/kubedns-deploy.yml.j2:10:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/ansible/templates/coredns-config.yml.j2:8:    addonmanager.kubernetes.io/mode: EnsureExists
roles/kubernetes-apps/ansible/templates/coredns-sa.yml.j2:9:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/ansible/templates/coredns-svc.yml.j2:10:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/ansible/templates/coredns-clusterrolebinding.yml.j2:9:    addonmanager.kubernetes.io/mode: EnsureExists
roles/kubernetes-apps/ansible/templates/kubedns-svc.yml.j2:10:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/efk/fluentd/templates/fluentd-config.yml.j2:9:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/efk/fluentd/templates/fluentd-ds.yml.j2:12:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/efk/elasticsearch/templates/efk-sa.yml:9:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/efk/elasticsearch/templates/efk-clusterrolebinding.yml:9:    addonmanager.kubernetes.io/mode: Reconcile
roles/kubernetes-apps/efk/elasticsearch/templates/elasticsearch-deployment.yml.j2:12:    addonmanager.kubernetes.io/mode: Reconcile
```